### PR TITLE
Add PHP 8.4 support

### PR DESCRIPTION
**Pull request by Slamdunk**

### CODE REVIEW

**Comment by Xerkus** [comment]
PHP 8.1 support won't end for one more year

**Comment by Slamdunk** [comment]
I didn't know Laminas has a committment towards all PHP versions with security support, fixed it.

BTW, I've opened this PR cause I have time for this right now: I'll fix everything only once PHP 8.4 is out.

**Comment by Xerkus** [comment]
We won't be adding PHP 8.4 to the versions just yet but we can merge patches for future compat.
To make CI run you can add php 8.4 to platform ignores. Check .laminas-ci.json history

**Comment by Slamdunk** [comment]
Don't know if I did everything you asked me, can you check it please?

**Comment by Xerkus** [comment]
Sorry if I got you confused. We do want 8.4 added to composer in the PR.  We won't be merging it until php reached stable enough milestone https://wiki.php.net/todo/php84

temporary override for the platform ignore is needed in .laminas-ci.json to make CI run checks with 8.4 as it otherwise currently fails at composer install step.

Changes here do not appear significant enough to do a separate release. They can wait with the PR.

**Comment by gsteel** [comment]
@Slamdunk 8.4 is available in CI now, could you refresh the lock file or rebase and re-push or something so that we can release this?

**Comment by Slamdunk** [comment]
Done

**Comment by Slamdunk** [comment]
Oh wait I need to add it to the CI

**Comment by Slamdunk** [comment]
https://github.com/laminas/laminas-hydrator/actions/runs/11817832480/job/32924094431#step:3:142

I guess we need to wait the other components then...

**Comment by gsteel** [comment]
Adding platform ignores to laminas-ci.json should do it: https://github.com/laminas/laminas-stratigility/pull/67/files#diff-3fb49ad00d3736f1ca8276f568f5d8bbe8c1604f6187ab5ecf921e1012f47c98

Also, upgrading and bumping dev deps too maybe

**Comment by Slamdunk** [comment]
All green now. Had to delete links to non-existent chapters in markdown files though

File: docs/book/v3/quick-start.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 The laminas-hydrator component provides functionality for hydrating objects (which is the act of populating an object from a set of data) and extracting data from them.
 
-The component contains [concrete implementations](#available_implementations) for a number of common use cases, such as by using arrays, object methods, and reflection, and provides [interfaces](#base_interfaces) for creating custom implementations.
+The component contains concrete implementations for a number of common use cases, such as by using arrays, object methods, and reflection, and provides interfaces for creating custom implementations.
```
Comment (froschdesign) [inline docs/book/v3/quick-start.md]: The first link can remain, but needs to be corrected:

```markdown
[concrete implementations](#available-implementations)
```

https://docs.laminas.dev/laminas-hydrator/v4/quick-start/#available-implementations

Comment (Slamdunk) [reply docs/book/v3/quick-start.md]: Done

**Review by gsteel — APPROVED** [review]
Thanks @Slamdunk 👍

**Pull request merged**